internal/controlplane/identity_events.go

```diff
@@ -148,24 +148,44 @@ func DeleteUser(ctx context.Context, store db.Store, authzClient authz.Client, u
 	defer store.Rollback(tx)
 	qtx := store.GetQuerierWithTransaction(tx)
 
-	user, err := qtx.GetUserBySubject(ctx, userId)
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			// user is already deleted
-			return nil
-		}
+	_, err = qtx.GetUserBySubject(ctx, userId)
+	// If the user doesn't exist, we still want to clean up any associated data
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		return fmt.Errorf("error retrieving user %v", err)
 	}
 
+	// Fetching the projects for user before the deletion was made.
+	// This allows us to clean up the project from the database
+	// if there are no more role assignments for the project.
+	projects, err := authzClient.ProjectsForUser(ctx, userId)
+	if err != nil {
+		return fmt.Errorf("error getting projects for user %v", err)
+	}
+
+	// We delete the user from the authorization system first
 	if err := authzClient.DeleteUser(ctx, userId); err != nil {
 		return fmt.Errorf("error deleting authorization tuple %v", err)
 	}
 
-	err = qtx.DeleteOrganization(ctx, user.OrganizationID)
-	if err != nil {
-		return fmt.Errorf("error deleting organization %w", err)
+	for _, proj := range projects {
+		// Given that we've deleted the user from the authorization system,
+		// we can now check if there are any role assignments for the project.
+		as, err := authzClient.AssignmentsToProject(ctx, proj)
```

Comment:
I think any role will do, we do want to clean up orphaned projects

Reply:
If we have a project which only has viewers on it, do we want to keep it?

What can the remaining members do?  Are they able to clean it up without deleting their own accounts?

Reply:
They can't do much on it, they'd have to add a support ticket for us to fix it. I think this is acceptable since we don't just want to randomly remove projects from folks.

Reply:
Okay, let's leave this for now.  I still think deleting the last admin should delete the project, but we can defer that to a later discussion.

Filed https://github.com/stacklok/minder/issues/2528 to avoid forgetting about this.